nova_vm/src/engine/bytecode/vm.rs

```diff
@@ -100,6 +100,26 @@ impl Vm {
         while let Some(instr) = executable.get_instruction(&mut vm.ip) {
             eprintln!("Executing instruction {:?}", instr.kind);
             match instr.kind {
+                Instruction::BitwiseNot => {
+                    // 2. Let oldValue be ? ToNumeric(? GetValue(expr)).
+                    let old_value = to_numeric(agent, vm.result.take().unwrap())?;
+
+                    // 3. If oldValue is a Number, then
+                    if let Ok(old_value) = Number::try_from(old_value) {
+                        // a. Return Number::bitwiseNOT(oldValue).
+                        vm.result = Some(Number::bitwise_not(agent, old_value)?.into());
+                    }
+                    // 4. Else,
+                    else {
+                        // a. Assert: oldValue is a BigInt.
+                        let Ok(old_value) = BigInt::try_from(old_value) else {
+                            unreachable!();
+                        };
+
+                        // b. Return BigInt::bitwiseNOT(oldValue).
+                        vm.result = Some(BigInt::bitwise_not(agent, old_value).into());
```

Comment:
**nitpick:** Consider using `.into_value()` explicitly. Doesn't really matter but I believe that's what we've generally been doing in the `vm.result` assignments.